dev/benchmarks/macrobenchmarks/lib/src/opacity_peephole.dart

```diff
@@ -0,0 +1,303 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import '../common.dart';
+
+// Various tests to verify that the Opacity layer propagates the opacity to various
+// combinations of children that can apply it themselves.
+// See https://github.com/flutter/flutter/issues/75697
+class OpacityPeepholePage extends StatelessWidget {
+  const OpacityPeepholePage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Opacity Peephole tests')),
+      body: ListView(
+        key: const Key('${kOpacityPeepholeRouteName}_listview'),
+        children: <Widget>[
+          for (OpacityPeepholeCase variant in allOpacityPeepholeCases)
+            ElevatedButton(
+              key: Key(variant.route),
+              child: Text(variant.name),
+              onPressed: () {
+                Navigator.pushNamed(context, variant.route);
+              },
+            )
+        ],
+      ),
+    );
+  }
+}
+
+typedef TestBuilder = Widget Function(double v);
+
+double _opacity(double v) => v * 0.5 + 0.25;
+int _red(double v) => (v * 255).round();
+int _green(double v) => _red(1 - v);
+int _blue(double v) => 0;
+
+class OpacityPeepholeCase {
+  OpacityPeepholeCase({required this.route, required this.name, required this.builder});
+
+  final String route;
+  final String name;
+  final TestBuilder builder;
+
+  Widget build(BuildContext context) {
+    return VariantPage(variant: this);
+  }
+}
+
+List<OpacityPeepholeCase> allOpacityPeepholeCases = <OpacityPeepholeCase>[
```

Comment:
This is a list of all of the available tests to look at for suggesting new cases or making a call on which cases are important to benchmark...